### DISCUSSION TITLE
fluids passing through solid boundary

### DISCUSSION THREAD

**Discussion by hosseinrezaa** [InteractiveComputerGraphics/SPlisHSPlasH#159] [Q&A]
Hi
I am using Akinci et. al 2012 boundary handling for simulation. Fluid particles can recognize the solid boundary properly. But, sometimes fluid particles passes through the solid boundary which contains numerous particles. What is the reason? How should we prevent fluid particles to penetrate the solid object?
Regards

**Comment by janbender** [comment]
If particles pass through the boundary, typically your time step size is too large. This could happen when you have fast moving particles. You could decrease the CFL factor to solve this.

**Reply by KimYoungGR** [reply]
![image](https://user-images.githubusercontent.com/85207792/213658785-f174e0d5-e4ac-4624-b995-4200b159219c.png)
Hi Mr. Bender
In the first project I solved the problem of particles passing through the wall by increasing the resolution, but in the second project this method did not work.
The geometry of the second program consists of several rooms of a factory building on one floor. This is more complicated than the first project . In this room the particles seem to be attracted by something and pass through the walls. 
I am using the volume map boundary condition.
![image](https://user-images.githubusercontent.com/85207792/213659910-138b2a24-4e8c-4ad6-96f1-92be3ed02748.png)

**Reply by janbender** [reply]
This looks like the map pushes them in the wrong direction. Either the map is just not inverted or the generation of the map failed. To generate the map you need a water-tight manifold mesh.

**Comment by KimYoungGR** [comment]
Hi
I meet the same problem. Do you sovle it now?

**Comment by janbender** [comment]
Can you give me more info about your scene configuration?

**Reply by KimYoungGR** [reply]
Hi Mr. Bender
Thanks for your reply, I tried to set a higher mapResolution and the problem seems solved.
I'm a beginner of SPH and I have a basic question:  In SPlisHSPlasH, does a single particle volunm equal to sphere that Radius is particleRadius?
